src/main/java/org/wso2/carbon/connector/connection/EmailConnectionManager.java

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.connector.connection;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.connector.exception.EmailConnectionException;
+import org.wso2.carbon.connector.exception.EmailConnectionPoolException;
+import org.wso2.carbon.connector.pojo.ConnectionConfiguration;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * Manages email connections and connection pools
+ */
+public class EmailConnectionManager {
+
+    private static final Log log = LogFactory.getLog(EmailConnectionManager.class);
+
+    private Map<String, EmailConnection> connectionMap;
+    private Map<String, EmailConnectionPool> connectionPoolMap;
+
+    private static EmailConnectionManager manager;
+
+    private EmailConnectionManager(){
+        this.connectionMap = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Is there a requirement to have synchronized map? Cannot we use concurrentHashMap? https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html

Reply:
Fixed